app/services/status_message_service.rb

```diff
@@ -0,0 +1,86 @@
+class StatusMessageService
+  include Rails.application.routes.url_helpers
+
+  attr_reader :status_message
+
+  def initialize(params, user)
+    @user = user
+    normalize_aspect_ids(params)
+    normalize_public_flag!(params)
+    @destination_aspect_ids = destination_aspect_ids(params)
+    @status_message = user.build_post(:status_message, params[:status_message])
+
+    build_location(params)
+    build_poll(params)
+    build_photos(params)
+    build_services(params)
+  end
+
+  def save
+    @status_message.save
+  end
+
+  def process_status
```

Comment:
`process_status_message` isn't all that long actually and might be the best compromise after all. Though given the class is already `StatusMessageService`, a plain `process` would actually be fine with me too.